File: EduStack-Backend/src/entities/Progress.ts
export interface Progress {
  id: string;
  userId: string;
  courseId: string;
  lessonId?: string;
  progress: number; // percentage
  completedAt?: Date;
  createdAt: Date;
  updatedAt: Date;
}

export interface CreateProgressData {
  userId: string;
  courseId: string;
  lessonId?: string;
  progress: number;
}

export interface UpdateProgressData {
  progress?: number;
  completedAt?: Date;
}

export interface CourseProgress {
  courseId: string;
  courseTitle: string;
  totalLessons: number;
  completedLessons: number;
  progress: number;
  lastAccessed?: Date;
}

File: EduStack-Backend/EduStack.Core/Entities/Progress.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EduStack.Core.Entities;

public class Progress
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid UserId { get; set; }

    [Required]
    public Guid CourseId { get; set; }

    public Guid? LessonId { get; set; }

    [Required]
    [Range(0, 100)]
    public int ProgressPercentage { get; set; } = 0; // percentage

    public DateTime? CompletedAt { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation Properties
    [ForeignKey("UserId")]
    public virtual User User { get; set; } = null!;

    [ForeignKey("CourseId")]
    public virtual Course Course { get; set; } = null!;

    [ForeignKey("LessonId")]
    public virtual Lesson? Lesson { get; set; }
}
